src/main/java/com/google/cloud/tools/gradle/appengine/appyaml/AppEngineAppYamlPlugin.java

```diff
@@ -89,10 +91,10 @@ private void configureExtensions() {
           // we can only set the default location of "archive" after project evaluation (callback)
           if (stageExtension.getArtifact() == null) {
             if (project.getPlugins().hasPlugin(WarPlugin.class)) {
-              War war = (War) project.getProperties().get("war");
+              War war = project.getTasks().withType(War.class).getByName("war");
```

Comment:
For some reason that I don't know, the previous code was returning null.

Also I think the new code may still unnecessarily create the `war` task.